include/boost/beast/experimental/core/uri/buffer.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2018 oneiric (oneiric at i2pmail dot org)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_CORE_URI_BUFFER_HPP
+#define BOOST_BEAST_CORE_URI_BUFFER_HPP
+
+#include <vector>
+#include <boost/beast/core/string.hpp>
+#include <boost/beast/experimental/core/uri/error.hpp>
+#include <boost/beast/experimental/core/uri/parts.hpp>
+
+namespace boost {
+namespace beast {
+namespace uri   {
+
+class buffer : public parts {
+  using const_iterator = std::vector<char>::const_iterator;
+  using value_type = std::vector<char>::value_type;
+  using size_type = std::vector<char>::size_type;
```

Comment:
Shouldn't these nested types be public?

Reply:
I didn't see a need to expose them publicly, but I'll move them.